src/test/compile-fail/dropck-object-cycle.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
```

Comment:
This is somewhat similar to the #26656 trait-object example. The `fn` example remains unsolved, through.